components/dashboard/src/workspaces/Workspaces.tsx

```diff
@@ -40,6 +43,8 @@ export default function () {
         })();
     }, [teams, location]);
 
+    const hasPreferredIde = typeof user?.additionalData?.ideSettings?.defaultIde !== "undefined";
```

Comment:
The first time I visit Perferences, `code` not selected. maybe we can init `hasPreferredIde`  before migrate (or make user setting copied) so that it’ll not affect your checking conditions and display of `SelectIDE`

Reply:
we could show "code" as default but I am concerned that it creates more confusion, considering that if you are a new user, in the homepage you would see the same component without any default, but in Preferences, if you do this, you would see "Code" as your default.. 🤔 

any more thoughts from @loujaybee @gtsiolis?

Reply:
We can show both `dashboard` and `preference` with default `code`

Reply:
Also, we can provide a interface in `User` namespace and implement it, so that others (me 🙈) can reuse the same logic.

We can add logic in `isOnboarding` if @loujaybee want to change onboarding condition
```ts
export namespace User {
    ...
    export function hasPerferredIde(user: User) {
        return !user?.additionalData?.ideSettings?.defaultIde;
    }
    export function isOnboarding(user: User) {
        return !hasPerferredIde(user);
    }
}
```

https://github.com/gitpod-io/gitpod/commit/24630fcacd7278bd1c047eb640d712a15a9065e9#diff-957d9dd15b947e3969539ebd0dabcb8b40ce8b5564e5820cd9bb27d88be1dbc7R98-R103

Reply:
I think it's very confusing if we show a default while we ask for a selection

Reply:
personally, I would leave unselected on the first screen, and, when we update the copy on the Preferences page, we can clarify that VS Code browser will be the default

Reply:
Same here https://github.com/gitpod-io/gitpod/pull/9432#discussion_r858715892

> I think we should show `code` as the default value on both the `Preference` and `Dashboard` pages, because without actually making any selections, workspaces will start with code, code is the real default. (code as the default option without modifying the user setting in DB)
> 
> In this case, when we click `skip`, it is equivalent to the user choosing code by default. So there is no need to add `skipIdeSelection`.
> 
> If we want to know how many people chose `skip`, we should add a `track event`